File: C2024/code_q1/data/build_parameters.py
"""构建未来七年复用的亩产、成本和价格参数。"""

from __future__ import annotations

import pandas as pd


def build_parameters(statistics: pd.DataFrame) -> tuple[pd.DataFrame, pd.DataFrame]:
    """补齐指南明确的“智慧大棚第一季=普通大棚第一季”参数继承。"""
    parameters = statistics.copy()
    ordinary_first = parameters.loc[
        (parameters["land_type"] == "普通大棚") & (parameters["season"] == "第一季")
    ].copy()
    inherited = ordinary_first.copy()
    inherited["land_type"] = "智慧大棚"
    inherited["parameter_source"] = "由普通大棚第一季按附件2注释继承"
    parameters["parameter_source"] = "附件2原始统计记录"
    parameters = pd.concat([parameters, inherited], ignore_index=True)
    return parameters, inherited

File: C2024/code_q1_matlab/data/build_parameters.m
function [parameters, inherited] = build_parameters(statistics)
%BUILD_PARAMETERS 将智慧大棚第一季参数按附件说明继承自普通大棚。

statistics.parameter_source = repmat({'附件2原始统计记录'}, height(statistics), 1);
ordinaryFirst = statistics(strcmp(statistics.land_type, '普通大棚') & strcmp(statistics.season, '第一季'), :);
inherited = ordinaryFirst;
inherited.land_type = repmat({'智慧大棚'}, height(inherited), 1);
inherited.parameter_source = repmat({'由普通大棚第一季按附件2注释继承'}, height(inherited), 1);
parameters = [statistics; inherited];
end



